libbeat/kibana/transform.go

```diff
@@ -7,113 +7,170 @@ import (
 	"github.com/elastic/beats/libbeat/common"
 )
 
-func TransformFields(timeFieldName string, title string, commonFields common.Fields) common.MapStr {
-	fields := []common.MapStr{}
-	fieldFormatMap := common.MapStr{}
-	keys := common.MapStr{}
+type Transformer struct {
+	fields                    common.Fields
+	transformedFields         []common.MapStr
+	transformedFieldFormatMap common.MapStr
+	timeFieldName             string
+	title                     string
+	keys                      common.MapStr
+}
+
+func NewTransformer(timeFieldName, title string, fields common.Fields) Transformer {
+	return Transformer{
+		fields:                    fields,
+		timeFieldName:             timeFieldName,
+		title:                     title,
+		transformedFields:         []common.MapStr{},
+		transformedFieldFormatMap: common.MapStr{},
+		keys: common.MapStr{},
+	}
+}
 
-	transformFields(keys, commonFields, &fields, fieldFormatMap, "")
+func (t *Transformer) TransformFields() common.MapStr {
+	t.transformFields(t.fields, "")
 
 	// add some meta fields
-	truthy := true
 	falsy := false
-	add(common.Field{Path: "_id", Type: "keyword", Index: &falsy, Analyzed: &falsy, DocValues: &falsy, Searchable: &falsy, Aggregatable: &falsy}, &fields, fieldFormatMap)
-	add(common.Field{Path: "_type", Type: "keyword", Index: &falsy, Analyzed: &falsy, DocValues: &falsy, Searchable: &truthy, Aggregatable: &truthy}, &fields, fieldFormatMap)
-	add(common.Field{Path: "_index", Type: "keyword", Index: &falsy, Analyzed: &falsy, DocValues: &falsy, Searchable: &falsy, Aggregatable: &falsy}, &fields, fieldFormatMap)
-	add(common.Field{Path: "_score", Type: "integer", Index: &falsy, Analyzed: &falsy, DocValues: &falsy, Searchable: &falsy, Aggregatable: &falsy}, &fields, fieldFormatMap)
+	t.add(common.Field{Path: "_id", Type: "keyword", Index: &falsy})
+	t.add(common.Field{Path: "_type", Type: "keyword", Index: &falsy})
```

Comment:
@simianhacker if a field has `index: false` can it then be `aggregatable` or `searchable`? 
So far the field `_type` had set `index: false, searchable: true, aggregatable: true`with `type: keyword`.
Afaik this does not make sense, so I changed it to `searchable: false, aggregatable: false` - but need your input here.